### DISCUSSION TITLE
Can maddy be run as root?

### DISCUSSION THREAD

**Discussion by shroff** [emersion/maddy#443] [Q&A] [answered]
I want to run maddy in a rootless Podman container, under a dedicated user.

The easieast way to manager file/directory ownership and permissions is to run the binary as root inside the container, since it maps to the user running the container on the host.

Is there a way to specify which user maddy should run as?

**Comment by foxcpp** [comment, marked as answer]
maddy will run as whatever user you start it as. There is no uid/gid switching or privilege dropping anywhere.
Provided Docker image is set to run it as root already, though there is an issue that suggests changing it: https://github.com/foxcpp/maddy/issues/323 (https://github.com/hexops/dockerfile#run-as-a-non-root-user).

**Comment by shroff** [comment]
Thanks for the clarification! The installation guide mentions in several places that "maddy never runs as root", so I assumed there was some sort of uid switching going on, but I wasn't able to find anything in the code that does that.
